examples/flutter_gallery/lib/demo/material/reorderable_list_demo.dart

```diff
@@ -0,0 +1,192 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+enum _ReorderableListType {
+  /// A list tile that contains a [CircleAvatar].
+  horizontalAvatar,
+
+  /// A list tile that contains a [CircleAvatar].
+  verticalAvatar,
+
+  /// A list tile that contains three lines of text and a checkbox.
+  threeLine,
+}
+
+class ReorderableListDemo extends StatefulWidget {
+  const ReorderableListDemo({ Key key }) : super(key: key);
+
+  static const String routeName = '/material/reorderable-list';
+
+  @override
+  _ListDemoState createState() => new _ListDemoState();
+}
+
+class _ListItem {
+  _ListItem(this.value, this.checkState);
+
+  final String value;
+
+  bool checkState;
+}
+
+class _ListDemoState extends State<ReorderableListDemo> {
+  static final GlobalKey<ScaffoldState> scaffoldKey = new GlobalKey<ScaffoldState>();
+
+  PersistentBottomSheetController<Null> _bottomSheet;
+  _ReorderableListType _itemType = _ReorderableListType.threeLine;
+  bool _reverseSort = false;
+  final List<_ListItem> _items = <String>[
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N',
+  ].map((String item) => new _ListItem(item, false)).toList();
+
+  void changeItemType(_ReorderableListType type) {
+    setState(() {
+      _itemType = type;
+    });
+    _bottomSheet?.setState(() { });
+    _bottomSheet.close();
+  }
+
+  void _showConfigurationSheet() {
+    final PersistentBottomSheetController<Null> bottomSheet = scaffoldKey.currentState.showBottomSheet((BuildContext bottomSheetContext) {
+      return new DecoratedBox(
+        decoration: const BoxDecoration(
+          border: const Border(top: const BorderSide(color: Colors.black26)),
+        ),
+        child: new ListView(
+          shrinkWrap: true,
+          primary: false,
+          children: <Widget>[
+            new RadioListTile<_ReorderableListType>(
+              dense: true,
+              title: const Text('Horizontal Avatars'),
+              value:_ReorderableListType.horizontalAvatar,
+              groupValue: _itemType,
+              onChanged: changeItemType,
+            ),
+            new RadioListTile<_ReorderableListType>(
+              dense: true,
+              title: const Text('Vertical Avatars'),
+              value: _ReorderableListType.verticalAvatar,
+              groupValue: _itemType,
+              onChanged: changeItemType,
+            ),
+            new RadioListTile<_ReorderableListType>(
+              dense: true,
+              title: const Text('Three-line'),
+              value: _ReorderableListType.threeLine,
+              groupValue: _itemType,
+              onChanged: changeItemType,
+            ),
+          ],
+        ),
+      );
+    });
+
+    setState(() {
+      _bottomSheet = bottomSheet;
+    });
+
+    // Garbage collect the bottom sheet when it closes.
+    bottomSheet.closed.whenComplete(() {
+      if (mounted) {
+        setState(() {
```

Comment:
Why is a _setState needed here?

Reply:
There was a comment I had left that's hidden by the commit history now.

I applied the change, but not setting state led to strange behavior: the menu in the corner doesn't update to reflect that you can open the bottom sheet again.